handler/oauth2/flow_generic_code_token.go

```diff
@@ -0,0 +1,244 @@
+// Copyright © 2024 Ory Corp
+// SPDX-License-Identifier: Apache-2.0
+
+package oauth2
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+
+	"github.com/ory/fosite/storage"
+
+	"github.com/ory/x/errorsx"
+
+	"github.com/ory/fosite"
+)
+
+// CodeTokenEndpointHandler handles the differences between Authorize code grant and extended grant types.
+type CodeTokenEndpointHandler interface {
```

Comment:
I found this interface name confusing (from the upstream pull request). But I did not manage to come up with a better name. Please let me know what you think.

Reply:
i'd say leave it, a lot of the names sound confusing to me but i guess it might be due to being not familiar enough with the codebase

for the time being, i guess it's best going with the flow and leave as is